main.tf

```diff
@@ -86,27 +63,25 @@ data "aws_route53_zone" "domain" {
 
 data "template_file" "user_data" {
   count    = module.this.enabled ? 1 : 0
-  template = file("${path.module}/user_data.sh")
+  template = file("${path.module}/${var.user_data_template}")
 
   vars = {
-    user_data       = join("\n", var.user_data)
-    welcome_message = module.this.stage
-    hostname        = "${module.this.name}.${join("", data.aws_route53_zone.domain.*.name)}"
-    search_domains  = join("", data.aws_route53_zone.domain.*.name)
-    ssh_user        = var.ssh_user
+    user_data   = join("\n", var.user_data)
+    ssm_enabled = var.ssm_enabled
+    ssh_user    = var.ssh_user
   }
 }
 
 resource "aws_instance" "default" {
   #bridgecrew:skip=BC_AWS_PUBLIC_12: Skipping `EC2 Should Not Have Public IPs` check. NAT instance requires public IP.
   #bridgecrew:skip=BC_AWS_GENERAL_31: Skipping `Ensure Instance Metadata Service Version 1 is not enabled` check until BridgeCrew support condition evaluation. See https://github.com/bridgecrewio/checkov/issues/793
   count         = module.this.enabled ? 1 : 0
-  ami           = var.ami
+  ami           = data.aws_ami.default.id
```

Comment:
if this is to run on Amazon linux, the SSM agent is preinstalled and starts at boot so there is no need to install it in or start in in userdata

Reply:
That's a good point! In this case, I think we can keep it because we have a feature flag to disable it, which should then stop it if it's running.